--- vmci-only/linux/driver.c.orig	2017-02-14 00:06:36.578515838 +0900
+++ vmci-only/linux/driver.c	2017-02-14 00:06:39.504534272 +0900
@@ -1468,12 +1468,16 @@
    int retval;
 
    down_read(&current->mm->mmap_sem);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
+   retval = get_user_pages(addr, 1, 0, &page, NULL);
+#else
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 6, 0)
    retval = get_user_pages(addr, 1, 1, 0, &page, NULL);
 #else
    retval = get_user_pages(current, current->mm, addr,
                            1, 1, 0, &page, NULL);
 #endif
+#endif
    up_read(&current->mm->mmap_sem);
 
    if (retval != 1) {
