--- vmci-only/shared/vm_basic_defs.h.orig	2017-05-16 03:28:26.905805655 +0900
+++ vmci-only/shared/vm_basic_defs.h	2017-05-16 03:29:05.857892195 +0900
@@ -719,16 +719,6 @@
 #define NONCLIENTMETRICSINFO_V1_SIZE CCSIZEOF_STRUCT(NONCLIENTMETRICS, \
                                                      lfMessageFont)
 
-/* This is not intended to be thread-safe. */
-#define DO_ONCE(code)                                                   \
-   do {                                                                 \
-      static Bool _doOnceDone = FALSE;                                  \
-      if (UNLIKELY(!_doOnceDone)) {                                     \
-         _doOnceDone = TRUE;                                            \
-         code;                                                          \
-      }                                                                 \
-   } while (0)
-
 /*
  * Bug 827422 and 838523.
  */
