diff -Nur wide-dhcpv6-20080615.orig/common.c wide-dhcpv6-20080615/common.c
--- wide-dhcpv6-20080615.orig/common.c	2014-05-29 00:14:47.810401494 +0900
+++ wide-dhcpv6-20080615/common.c	2014-05-29 00:15:38.833932148 +0900
@@ -3205,7 +3205,7 @@
 		}
 	}
 }
-
+#if 0
 void
 dprintf(int level, const char *fname, const char *fmt, ...)
 {
@@ -3239,7 +3239,7 @@
 	} else
 		syslog(level, "%s%s%s", fname, printfname ? ": " : "", logbuf);
 }
-
+#endif
 int
 ifaddrconf(cmd, ifname, addr, plen, pltime, vltime)
 	ifaddrconf_cmd_t cmd;
diff -Nur wide-dhcpv6-20080615.orig/common.h wide-dhcpv6-20080615/common.h
--- wide-dhcpv6-20080615.orig/common.h	2014-05-29 00:14:47.803401558 +0900
+++ wide-dhcpv6-20080615/common.h	2014-05-29 00:15:03.886253529 +0900
@@ -155,7 +155,7 @@
 extern int in6_addrscopebyif __P((struct in6_addr *, char *));
 extern int in6_scope __P((struct in6_addr *));
 extern void setloglevel __P((int));
-extern void dprintf __P((int, const char *, const char *, ...));
+/*extern void dprintf __P((int, const char *, const char *, ...)); */
 extern int get_duid __P((char *, struct duid *));
 extern void dhcp6_init_options __P((struct dhcp6_optinfo *));
 extern void dhcp6_clear_options __P((struct dhcp6_optinfo *));
